sdk/data/azcosmos/example_test.go

```diff
@@ -0,0 +1,161 @@
+
+	resp, err := database.DatabaseProperties.Database.CreateContainer(ctx, properties, throughput, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	container := resp.ContainerProperties.Container
+
+	// ===== 3. Update container properties =====
+
+	resp, err = container.Read(ctx, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	updatedProperties := CosmosContainerProperties{
+		Id: "aContainer",
+		PartitionKeyDefinition: PartitionKeyDefinition{
+			Paths: []string{"/myPartitionKey"},
+		},
+		ETag: "someEtag",
+		IndexingPolicy: &IndexingPolicy{
+			IncludedPaths: []IncludedPath{},
+			ExcludedPaths: []ExcludedPath{},
+			Automatic:     false,
+			IndexingMode:  IndexingModeNone,
+		},
+	}
+
+	// Replace container properties
+	resp, _ = container.Replace(ctx, updatedProperties, nil)
+
+	// Read the manual throughput property
+	throughputResponse, _ := container.ReadThroughput(ctx, nil)
+	_, _ = throughputResponse.ThroughputProperties.ManualThroughput()
+
+	// Replace manual throughput property
+	newScale := NewManualThroughputProperties(500)
+	_, _ = container.ReplaceThroughput(ctx, *newScale, nil)
+
+	// Migrate from manual throughput to autoscale
+	_, _ = throughputResponse.ThroughputProperties.AutoscaleMaxThroughput()
+
+	newScale = NewAutoscaleThroughputProperties(10000)
+	_, _ = container.ReplaceThroughput(ctx, *newScale, nil)
+
+	// ===== 4. Item CRUD =====
+
+	// Items in an Azure Cosmos container are uniquely identified by their id and partition key value.
+	// pk, _ := NewPartitionKey("myPartitionKey")
+
+	item := map[string]string{
+		"id":    "1",
+		"value": "2",
+	}
+
+	// Create item.
+	itemResponse, err := container.CreateItem(ctx, nil, item, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Read item.
+	itemResponse, err = container.ReadItem(ctx, nil, "1", nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Item response body
+	var itemResponseBody map[string]interface{}
+	err = json.Unmarshal(itemResponse.Value, &itemResponseBody)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// ===== 5. Session consistency =====
+
+	itemResponse, err = container.UpsertItem(ctx, nil, item, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	itemSessionToken := itemResponse.SessionToken()
+	itemResponse, err = container.ReadItem(ctx, nil, "1", &CosmosItemRequestOptions{SessionToken: itemSessionToken})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// ===== 6. Optimistic Concurrency Etag PreConditionFail =====
+
+	// Azure Cosmos DB supports optimistic concurrency control to prevent lost updates or deletes and detection of conflicting operations.
+	itemResponse, err = container.ReadItem(ctx, nil, "1", nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	itemEtag := itemResponse.ETag()
+	item = map[string]string{
+		"id":    "1",
+		"value": "newValue",
+	}
+
+	// Check the item response status code. If an error is imitted and the response code is 412 then retry operation.
+	for itemResponse.RawResponse.StatusCode == 412 {
```

Comment:
Based on this logic, the for will never execute. `itemResponse` is coming from the ReadItem with an HTTP 200 response